File: ScaleformUI_Csharp/Menus/Pause Menus/PauseMenu/Items/SettingsItems/SettingsSliderItem.cs
using ScaleformUI.Elements;

namespace ScaleformUI.PauseMenu
{
    public delegate void SettingsSliderEvent(SettingsSliderItem item, int value);
    public class SettingsSliderItem : SettingsItem
    {
        internal int _value;
        private SColor coloredBarColor = SColor.HUD_Freemode;
        public event SettingsSliderEvent OnBarChanged;
        public event SettingsSliderEvent OnSliderSelected;
        public int MaxValue { get; set; } = 100;
        public int Value
        {
            get => _value;
            set
            {
                _value = value;
                if (ParentColumn != null && ParentColumn.visible)
                    ParentColumn.UpdateSlot(ParentColumn.Items.IndexOf(this));
                OnBarChanged?.Invoke(this, _value);
            }
        }
        public SColor ColoredBarColor
        {
            get => coloredBarColor;
            set
            {
                coloredBarColor = value;
                if (ParentColumn != null && ParentColumn.visible)
                    ParentColumn.UpdateSlot(ParentColumn.Items.IndexOf(this));
            }
        }

        public SettingsSliderItem(string label, int max, int startIndex, SColor barColor) : base(label, "")
        {
            ItemType = SettingsItemType.SliderBar;
            Label = label;
            MaxValue = max;
            Value = startIndex;
            ColoredBarColor = barColor;
        }

        public void SliderChanged()
        {
            OnBarChanged?.Invoke(this, Value);
        }

        public void SliderSelected()
        {
            OnSliderSelected?.Invoke(this, Value);
        }
    }
}

File: ScaleformUI_Lua/src/Menus/PauseMenus/PauseMenu/Items/SettingsItems/SettingsSliderItem.lua
SettingsSliderItem = {}
SettingsSliderItem.__index = SettingsSliderItem
setmetatable(SettingsSliderItem, { __index = SettingsItem })
SettingsSliderItem.__call = function() return "SettingsSliderItem" end

---@class SettingsSliderItem
---@field public Base PauseMenuItem
---@field public ItemType SettingsItemType
---@field public Label string
---@field public MaxValue number
---@field public Parent PauseMenuItem
---@field public OnBarChanged fun(item: SettingsSliderItem, value: number)
---@field public OnSliderSelected fun(item: SettingsSliderItem, value: number)

---Creates a new SettingsSliderItem.
---@param label string
---@param max number
---@param startIndex number
---@param barColor SColor
---@return table
function SettingsSliderItem.New(label, max, startIndex, barColor)
    local base = SettingsItem.New(label, "")
    base.ItemType = SettingsItemType.SliderBar
    base.MaxValue = max
    base._value = startIndex
    base._coloredBarColor = barColor or SColor.HUD_Freemode
    base.OnBarChanged = function(item, value)
    end
    base.OnSliderSelected = function(item, value)
    end
    return setmetatable(base, SettingsSliderItem)
end

---Set the value of the item.
---@param value number
---@return number
function SettingsSliderItem:Value(value)
    if value ~= nil then
        self._value = value
        if self.ParentColumn ~= nil and self.ParentColumn:visible() then
            self.ParentColumn:UpdateSlot(IndexOf(self.ParentColumn.Items, self))
        end
        self.OnBarChanged(self, value)
    end
    return self._value
end

---Set the color of the colored bar.
---@param color SColor
---@return SColor
function SettingsSliderItem:ColoredBarColor(color)
    if color ~= nil then
        self._coloredBarColor = color
        if self.ParentColumn ~= nil and self.ParentColumn:visible() then
            self.ParentColumn:UpdateSlot(IndexOf(self.ParentColumn.Items, self))
        end
    end
    return self._coloredBarColor
end
